subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -456,35 +456,41 @@ uint8_t ll_df_set_cl_iq_sampling_enable(uint16_t handle,
 #endif /* CONFIG_BT_CTLR_DF_DEBUG_ENABLE */
 
 		/* Enable of already enabled CTE updates AoA configuration */
-		if (!((IS_ENABLED(CONFIG_BT_CTLR_DF_ANT_SWITCH_1US) &&
-		       slot_durations == BT_HCI_LE_ANTENNA_SWITCHING_SLOT_1US) ||
-		      slot_durations == BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US)) {
-			return BT_HCI_ERR_UNSUPP_FEATURE_PARAM_VAL;
+
+		/* According to Core 5.3 Vol 4, Part E, section 7.8.82 slot_durations,
+		 * switch_pattern_len and ant_ids are used only for AoA and do not affect
+		 * reception of AoD CTE. If AoA is not supported relax command validation
+		 * to improve interoperability with different Host implementations.
+		 */
+		if (IS_ENABLED(CONFIG_BT_CTLR_DF_ANT_SWITCH_RX)) {
+			if (!((IS_ENABLED(CONFIG_BT_CTLR_DF_ANT_SWITCH_1US) &&
+			       slot_durations == BT_HCI_LE_ANTENNA_SWITCHING_SLOT_1US) ||
+			      slot_durations == BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US)) {
+				return BT_HCI_ERR_UNSUPP_FEATURE_PARAM_VAL;
+			}
+
+			if (switch_pattern_len < BT_HCI_LE_SWITCH_PATTERN_LEN_MIN ||
+			    switch_pattern_len > BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN || !ant_ids) {
+				return BT_HCI_ERR_UNSUPP_FEATURE_PARAM_VAL;
+			}
+			memcpy(cfg->ant_ids, ant_ids, switch_pattern_len);
```

Comment:
Blank line after if-clause, and `(void)` for `memcpy`.

```suggestion

			(void)memcpy(cfg->ant_ids, ant_ids, switch_pattern_len);
```

Reply:
Corrected